### ISSUE TITLE
EntityModel.SetupAnim use the last bind entity Animation

### ISSUE THREAD

**Issue by MrScautHD** [MinecraftForge/MinecraftForge#8367]
**Minecraft Version:** 1.18.1

**Forge Version:** {Forge 39.0.10. 1.18.1}

**Logs:** i get no Crash or something

**Steps to Reproduce:**
i tried to play a animation if my entity on the ground i checked it with

if (entity.isGround) {
   Animation...
}

with just 1 entity does that work but if i spawn a second does the first entity use the same animations then the second entity

(i get this errors by the most entitys i had, just not by the extended version of the piglin, by the mogler... i get this bug)

Render Code:
https://pastebin.com/5pKiwGQM

Model Code:
https://pastebin.com/YzMk1AGM

Entity Code:
https://pastebin.com/THezZkui

Video of the bug:
https://user-images.githubusercontent.com/65916181/148656941-c016037a-a1b3-431e-aa3c-380c261d5023.mp4

i really don't can understand the problem, but im sure it is a bug of mineceaft or forge

**Comment by XFactHD** [comment]
Entity models are global objects and not per entity. If you set the animation parameters for a specific case but you don't set those parameters to default for when that specific case doesn't apply to the entity currently being rendered, then you will get this kind of result.

TLDR: this is not a bug in MC or Forge, it's a bug in your code.

PS: you should not be using `@OnlyIn` anywhere in your own code.

**Comment by MrScautHD** [comment]
yea i see it right not thx :)
